### PR TITLE
beta.2

### PR DESCRIPTION
**Pull request by tanner0101**

### CODE REVIEW

File: Sources/APNS/APNS.swift [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+                    pool.shutdown()
+                }
+            }
+        }
+
+        var storage: Storage {
+            if self.application.storage[Key.self] == nil {
+                self.initialize()
+            }
+            return self.application.storage[Key.self]!
```
Comment (kylebrowning) [inline Sources/APNS/APNS.swift]: Nit, but wouldn't this be cleaner?

```swift
        var storage: Storage {
            guard let existingStorage = self.application.storage[Key.self] else {
                self.initialize()
            }
            return existingStorage
        }
```

File: Sources/APNS/APNS.swift [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+
+extension Request {
+    public var apns: APNS {
+        .init(request: self)
+    }
+
+    public struct APNS {
+        let request: Request
+
+
```
Comment (kylebrowning) [inline Sources/APNS/APNS.swift]: nit: extra space

File: Sources/APNS/APNS.swift [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+            with encoder: JSONEncoder = JSONEncoder(),
+            expiration: Date? = nil,
+            priority: Int? = nil,
+            collapseIdentifier: String? = nil,
+            topic: String? = nil
+        ) -> EventLoopFuture<Void> {
+            self.send(BasicNotification(aps: payload), pushType: pushType, to: deviceToken, with: encoder, expiration: expiration, priority: priority, collapseIdentifier: collapseIdentifier, topic: topic)
+        }
+
+
```
Comment (kylebrowning) [inline Sources/APNS/APNS.swift]: nit: extra space.

File: Sources/APNS/APNS.swift (hunk 1 above)
Comment (tanner0101) [reply Sources/APNS/APNS.swift]: `guard` must return before it exits scope so that wouldn't work

Comment (kylebrowning) [reply Sources/APNS/APNS.swift]: sorry meant, 
```swift
        var storage: Storage {
            guard let existingStorage = self.application.storage[Key.self] else {
                return self.initialize()
            }
            return existingStorage
        }
```

but yeah, then initialize needs to return the key. So its just semantic.

**Pull request merged**